dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java

```diff
@@ -106,63 +118,74 @@ public void addTask4TimeoutCheck(TaskInstance taskInstance) {
             return;
         }
         if (TimeoutFlag.OPEN == taskDefinition.getTimeoutFlag()) {
-            taskInstanceTimeoutCheckList.put(taskInstance.getId(), taskInstance.getProcessInstanceId());
+            taskInstanceTimeoutCheckList.put(taskInstance.getTaskCode(), taskInstance.getProcessInstanceId());
```

Comment:
Does the same task run in different process instances?

Reply:
I had fix it by `TaskInstanceKey`, please take a look again.